kerngen/high_parser/types.py

```diff
@@ -134,6 +134,12 @@ class EmptyLine(BaseModel):
     """Holder of an empty line"""
 
 
+# TODO remove hardcoding. Maybe move this to Config?
+NATIVE_POLY_SIZE = 8192
+MIN_POLY_SIZE = 8192
```

Comment:
```suggestion
MIN_POLY_SIZE = 16384
```
Although certain operations are correct for MIN_POLY_SIZE=8192, NTT and INTT minimum poly size is limited to 16384. Since the MIN_POLY_SIZE is checked globally, we have to respect that limitation.

Reply:
> Although certain operations are correct for MIN_POLY_SIZE=8192, NTT and INTT minimum poly size is limited to 16384. Since the MIN_POLY_SIZE is checked globally, we have to respect that limitation.

Would this limit us to not being able to run those other ops (add, mul, etc.) at the 8K poly size? Prior there were checks that allowed certain kernels to be run at 8K and others at 16K+

Reply:
Hi @kylanerace 
The conundrum is as follows: with the current implementations of the checks and to keep consistence across all the operations, yes it will limit us to 16K polynomials. RELIN operations will only work with min size 16K, due to dependency on NTT/INTT operations which are performed by the backend.

The question we have to ask ourselves is: Considering the FHE computation, would we be supporting operations that worked with 8K polynomials but not require and NTT/INTT?  I guess the answer is no, for the time being. Since the kernels are configurable as far as the parametrization is concerned, we can change it later should the backend support smaller polynomials